Require PHP-extensions via composer

Ruud-Zuiderlicht opened the pull request:
Composer allows to check requirements of the server. Currently the only way to check server requirements is by going to https://documentation.concrete5.org/developers/introduction/system-requirements and translate the list to some extensions and verify these are installed. I thought it might be beneficial  to add a more formal list of at least the minimal requirements.

For lack of better I used the list mentioned for concrete5 8.x, so this might be incomplete. I didn't include Mcrypt as I haven't got it installed anywhere and it is declared abandonware.
- PHP 7.4+
- MySQL (with PDO extensions)
- DOM
- SimpleXML
- iconv
- GD Library with Freetype
- Fileinfo
- Mbstring
- CURL
- _Mcrypt_
- ZipArchive (For automatic updates and community functionality)

Checking requirements is easy:
```bash
$ composer check-platform-reqs
composer-plugin-api  1.1.0       success  
ext-curl             7.4.7       success
ext-dom              20031129    success
ext-fileinfo         7.4.7       success
ext-gd               7.4.7       success
ext-iconv            7.4.7       success
ext-mbstring         7.4.7       success
ext-pdo_mysql        7.4.7       success
ext-SimpleXML        7.4.7       success
ext-zip              1.15.6      success
php                  7.4.7       success
```

Composer also allows to view installed extensions, though then you'd still need to manually compare to the list on the website:
` $ composer show -p`

aembler commented:
I think our current develop branch php requirement is actually 7.2

mlocati commented:
I'd also move such kind of changes to the `concrete/composer.json` file (which is also used by composer-based installations).

Ruud-Zuiderlicht commented:
I have moved the dependencies, though I'm not seeing `ext-pdo_mysql` when I test locally. I'm assuming this changes when this is required in concrete/composer.json online? @mlocati do you know if that is the case?

I have also changed the required PHP-version to 7.2. The table I used is somewhat confusing because it lists concrete5 versions that will work relative to PHP versions where I had read it as a minimum PHP-version for the stated c5 version.

Ruud-Zuiderlicht commented:
Because of the changes I also checked the PHP target 8.0. @aembler thinks it is feasible to obtain support https://github.com/concrete5/concrete5/issues/9314#issuecomment-769272964. I've kept PHP 8 as a valid platform target for now, but I can remove that until the updates are done?
